contracts/TokenHolderRevenueFund.sol

```diff
@@ -96,18 +90,31 @@ contract TokenHolderRevenueFund is Ownable, AccrualBeneficiary, Servable, SelfDe
     // Functions
     // -----------------------------------------------------------------------------------------------------------------
     function setRevenueTokenAddress(address newAddress) public onlyOwner notNullAddress(newAddress) {
-        address oldAddress;
+        if (newAddress != address(revenueToken)) {
+            address oldAddress;
 
-        if (newAddress != revenueToken) {
-            // Set new revenue token address
+            //set new revenue token address
             oldAddress = revenueToken;
-            revenueToken = newAddress;
+            revenueToken = RevenueToken(newAddress);
 
-            // Emit event
+            //emit event
             emit RevenueTokenChangedEvent(oldAddress, newAddress);
         }
     }
 
+    function setCurrencyManagerAddress(address newAddress) public onlyOwner notNullAddress(newAddress) {
```

Comment:
Although not changed in this PR I suggest this function signature be changed to `changeCurrencyManager(CurrencyManager newCurrencyManager) public` in order to fit the pattern used in other contracts.